### PR TITLE
ApmLoggerToNLog - Sample code optimized for NLog

### PR DESCRIPTION
**Pull request by snakefoot**
Ensure implementation of `IsEnabled` is working. Do not format log-message unless needed.

### CODE REVIEW

**Comment by snakefoot** [comment]
Cannot figure out how your sign-process works. Have signed it 3 times, and no change.

Please choose another sign-process-service, or improve the guidance.

File: sample/AspNetFullFrameworkSampleApp/App_Start/ApmLoggerToNLog.cs [hunk 1]
```diff
@@ -8,17 +8,33 @@ namespace AspNetFullFrameworkSampleApp
...
+		public ApmLoggerToNLog()
+			:this(string.Empty)
+		{
+
+		}
+
+		public ApmLoggerToNLog(string loggerName)
+		{
+			_logger = new Lazy<Logger>(() => LogManager.GetLogger(loggerName ?? string.Empty), true);
+		}
```
Comment (gregkalapos) [inline sample/AspNetFullFrameworkSampleApp/App_Start/ApmLoggerToNLog.cs]: nit: [we use](https://github.com/elastic/apm-agent-dotnet/blob/master/.editorconfig#L70) expression bodied .ctor for 1 liners

```suggestion
		public ApmLoggerToNLog(string loggerName) => _logger = new Lazy<Logger>(() => LogManager.GetLogger(loggerName ?? string.Empty), true);
```

Comment (snakefoot) [reply sample/AspNetFullFrameworkSampleApp/App_Start/ApmLoggerToNLog.cs]: Looks more convoluted , but your wish, my command.

**Comment by snakefoot** [comment]
Looks like the suggestion-commit changed all newline-characters. Not sure why.

**Comment by gregkalapos** [comment]
> Looks like the suggestion-commit changed all newline-characters. Not sure why.

Yeah, we struggle with "CRLF vs LF" hell. https://github.com/elastic/apm-agent-dotnet/issues/563 

I suspect the file was CRLF (because it was created on Windows) but now GitHub made it LF.

**Comment by snakefoot** [comment]
I can redo the commit suggestion without newlines change. If you want?

**Comment by gregkalapos** [comment]
> I can redo the commit suggestion without newlines change. If you want?

That'd be awesome. I looked at it and actually it's the other way - it seems it was LF (that's what we want) and became CRLF (which we don't like) - so if it was LF I'd prefer it to remain all LF. 

#563 should take care of this long term. Sorry about the trouble.

**Comment by snakefoot** [comment]
Forced commit executed

**Pull request merged**